sw/device/mask_rom/mask_rom.c

```diff
@@ -3,13 +3,43 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "sw/device/mask_rom/mask_rom.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
 #include "sw/device/lib/base/stdasm.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/rom_exts/rom_ext_manifest_parser.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+// This is the expected ROM_EXT Manifest Identifier
+static const uint32_t kRomExtIdentifierExpected = 0x4552544F;
+
+typedef void(boot_fn)(void);
```

Comment:
nit: Would it make sense to move these to the parser?

Reply:
I think it probably should be defined in this Mask ROM, as parser doesn't necessarily need to know the ROM_EXT boot function signature (not applicable now, but is a subject to change later).

Reply:
There are open questions about where we document the contract between the Mask ROM and the ROM_EXT. I want to leave this here until those are resolved (because they will influence the type of this function).